src/bcrypt_node.cc

```diff
@@ -63,86 +60,71 @@ bool ValidateSalt(const char* salt) {
-    }
-
-    ~SaltAsyncWorker() {}
+class SaltAsyncWorker : public Napi::AsyncWorker {
+    public:
+        SaltAsyncWorker(Napi::Function& callback, std::string seed, ssize_t rounds)
+            : AsyncWorker(callback), seed(seed), rounds(rounds) {
+        }
 
-    void Execute() {
-        char salt[_SALT_LEN];
-        bcrypt_gensalt(rounds, (u_int8_t *)&seed[0], salt);
-        this->salt = std::string(salt);
-    }
+        ~SaltAsyncWorker() {}
 
-    void HandleOKCallback() {
-        Nan::HandleScope scope;
+        void Execute() {
+            char salt[_SALT_LEN];
+            bcrypt_gensalt(rounds, (u_int8_t *)&seed[0], salt);
+            this->salt = std::string(salt);
+        }
 
-        Local<Value> argv[2];
-        argv[0] = Nan::Undefined();
-        argv[1] = Nan::Encode(salt.c_str(), salt.size(), Nan::BINARY);
-        callback->Call(2, argv);
-    }
+        void OnOK() {
+            Napi::HandleScope scope(Env());   
+            Callback().Call({Env().Undefined(), Napi::String::New(Env(), salt)});
+        }
 
-private:
-    std::string seed;
-    std::string salt;
-    ssize_t rounds;
+    private:
+        std::string seed;
+        std::string salt;
+        ssize_t rounds;
 };
 
-NAN_METHOD(GenerateSalt) {
-    Nan::HandleScope scope;
-
+Napi::Value GenerateSalt(const Napi::CallbackInfo& info) {
     if (info.Length() < 3) {
-        Nan::ThrowTypeError("3 arguments expected");
-        return;
+        throw Napi::TypeError::New(info.Env(), "3 arguments expected");
+        return info.Env().Undefined();
     }
-
-    if (!Buffer::HasInstance(info[1]) || Buffer::Length(info[1].As<Object>()) != 16) {
-        Nan::ThrowTypeError("Second argument must be a 16 byte Buffer");
-        return;
+    if (!info[1].IsBuffer() || (info[1].As<Napi::Buffer<char>>()).Length() != 16) {
+        throw Napi::TypeError::New(info.Env(), "Second argument must be a 16 byte Buffer");
+        return info.Env().Undefined();
     }
-
-    const int32_t rounds = Nan::To<int32_t>(info[0]).FromMaybe(0);
-    Local<Object> seed = info[1].As<Object>();
-    Local<Function> callback = Local<Function>::Cast(info[2]);
-
-    SaltAsyncWorker* saltWorker = new SaltAsyncWorker(new Nan::Callback(callback),
-        std::string(Buffer::Data(seed), 16), rounds);
-
-    Nan::AsyncQueueWorker(saltWorker);
+    const int32_t rounds = info[0].As<Napi::Number>();
+    Napi::Function callback = info[2].As<Napi::Function>();
+    Napi::Buffer<char> seed = info[1].As<Napi::Buffer<char>>();
+    SaltAsyncWorker* saltWorker = new SaltAsyncWorker(callback, std::string(seed.Data(), 16), rounds);
+    saltWorker->Queue();
+    return info.Env().Undefined();
 }
 
-NAN_METHOD(GenerateSaltSync) {
-    Nan::HandleScope scope;
-
+Napi::Value GenerateSaltSync (const Napi::CallbackInfo& info) {
     if (info.Length() < 2) {
-        Nan::ThrowTypeError("2 arguments expected");
-        return;
+        throw Napi::TypeError::New(info.Env(), "3 arguments expected");
```

Comment:
should be 2 args expected ?

Reply:
Ok I think that I’m the code there are some errors. I will change as soon as possible now I’m leaving from Vancouver to Italy and I cannot be very active